exp/operator/main.go

```diff
@@ -17,69 +17,176 @@ limitations under the License.
+	"k8s.io/klog"
 	"k8s.io/klog/klogr"
-	ctrl "sigs.k8s.io/controller-runtime"
-
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
 	operatorv1alpha4 "sigs.k8s.io/cluster-api/exp/operator/api/v1alpha4"
 	"sigs.k8s.io/cluster-api/exp/operator/controllers"
+	"sigs.k8s.io/cluster-api/feature"
 	"sigs.k8s.io/cluster-api/version"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
 	// +kubebuilder:scaffold:imports
 )
 
 var (
 	scheme   = runtime.NewScheme()
 	setupLog = ctrl.Log.WithName("setup")
+
+	// flags
+	metricsBindAddr             string
+	enableLeaderElection        bool
+	leaderElectionLeaseDuration time.Duration
+	leaderElectionRenewDeadline time.Duration
+	leaderElectionRetryPeriod   time.Duration
+	watchNamespace              string
+	watchFilterValue            string
+	profilerAddress             string
+	coreProviderConcurrency     int
+	syncPeriod                  time.Duration
+	webhookPort                 int
+	webhookCertDir              string
+	healthAddr                  string
 )
 
 func init() {
-	_ = clientgoscheme.AddToScheme(scheme)
+	klog.InitFlags(nil)
 
+	_ = clientgoscheme.AddToScheme(scheme)
 	_ = operatorv1alpha4.AddToScheme(scheme)
 	// +kubebuilder:scaffold:scheme
 }
 
+// InitFlags initializes the flags.
+func InitFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&metricsBindAddr, "metrics-bind-addr", ":8080",
+		"The address the metric endpoint binds to.")
+
+	fs.BoolVar(&enableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
+
+	fs.DurationVar(&leaderElectionLeaseDuration, "leader-elect-lease-duration", 15*time.Second,
+		"Interval at which non-leader candidates will wait to force acquire leadership (duration string)")
+
+	fs.DurationVar(&leaderElectionRenewDeadline, "leader-elect-renew-deadline", 10*time.Second,
+		"Duration that the leading controller manager will retry refreshing leadership before giving up (duration string)")
+
+	fs.DurationVar(&leaderElectionRetryPeriod, "leader-elect-retry-period", 2*time.Second,
+		"Duration the LeaderElector clients should wait between tries of actions (duration string)")
+
+	fs.StringVar(&watchNamespace, "namespace", "",
+		"Namespace that the controller watches to reconcile cluster-api objects. If unspecified, the controller watches for cluster-api objects across all namespaces.")
+
+	fs.StringVar(&watchFilterValue, "watch-filter", "",
+		fmt.Sprintf("Label value that the controller watches to reconcile cluster-api objects. Label key is always %s. If unspecified, the controller watches for all cluster-api objects.", clusterv1.WatchLabel))
+
+	fs.StringVar(&profilerAddress, "profiler-address", "",
+		"Bind address to expose the pprof profiler (e.g. localhost:6060)")
+
+	fs.IntVar(&coreProviderConcurrency, "core-provider-concurrency", 10,
```

Comment:
```suggestion
	fs.IntVar(&coreProviderConcurrency, "concurrency", 10,
```

Reply:
Same answer as to the previous comment. Do we want to follow the pattern from CAPI and each resource should have its own concurrency flag? In this case, `core-provider` represents a CRD.

Reply:
I think it's fine here to start with just a simple concurrency flag to be shared with all the CRDs

Reply:
ok, I fixed it